core/src/main/java/org/mobilitydata/gtfsvalidator/input/GtfsInput.java

```diff
@@ -56,11 +63,100 @@ public static GtfsInput createFromPath(Path path) throws IOException {
     }
     if (path.getFileSystem().equals(FileSystems.getDefault())) {
       // Read from a local ZIP file.
-      return new GtfsZipFileInput(new ZipFile(path.toFile()));
+      zipFile = new ZipFile(path.toFile());
+      if (containsSubfolderWithGtfsFile(path)) {
+        noticeContainer.addValidationNotice(
+            new InvalidInputFilesInSubfolderNotice(invalidInputMessage));
+      }
+      return new GtfsZipFileInput(zipFile);
     }
     // Load a remote ZIP file to memory.
-    return new GtfsZipFileInput(
-        new ZipFile(new SeekableInMemoryByteChannel(Files.readAllBytes(path))));
+    zipFile = new ZipFile(new SeekableInMemoryByteChannel(Files.readAllBytes(path)));
+    if (containsSubfolderWithGtfsFile(path)) {
+      noticeContainer.addValidationNotice(
+          new InvalidInputFilesInSubfolderNotice(invalidInputMessage));
+    }
+    return new GtfsZipFileInput(zipFile);
+  }
+
+  /**
+   * Check if a zip file contains a subfolder with GTFS files
+   *
+   * @param path
+   * @return
+   */
+  public static boolean containsSubfolderWithGtfsFile(Path path) {
+    boolean containsGtfsFileInSubfolder = false;
+    boolean containsSubfolder = false;
+    String subfolder = null;
+    try (ZipInputStream zipInputStream =
+        new ZipInputStream(new BufferedInputStream(new FileInputStream(path.toFile())))) {
+      ZipEntry entry;
+      while ((entry = zipInputStream.getNextEntry()) != null) {
+        String entryName = entry.getName();
+        if (entry.isDirectory()) {
+          containsSubfolder = true;
+          subfolder = entry.getName();
+        }
+        if (containsSubfolder && entryName.contains(subfolder) && entryName.endsWith(".txt")) {
+          String[] files = entryName.split("/");
+          String lastElement = files[files.length - 1];
+          if (containsGtfsFile(lastElement)) {
+            containsGtfsFileInSubfolder = true;
+          }
+        }
+      }
+    } catch (IOException ioException) {
+      logger.atSevere().withCause(ioException).log(
```

Comment:
It's ok, as explained in the above comment.

Reply:
Not sure I understand. What above comment?
I will rephrase my question: If there is an error reading the file (IOException) we don't really know if there is a subfolder or not. So maybe we should let the IOException through?
Or maybe just a comment explaining why we return false if there is an IO exception. Something like:
```
// Return false in case of IOException, meaning we will not issue a notice for that. 
// We expect a systems error notice to be issued for that error.
```